### PR TITLE
Fix assigning tensor with requires_grad as constant in export

### PR DESCRIPTION
**Pull request by tugsbayasgalan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #137997

When we insert cojstants into unlifted graph, we need to detach them if they require grad BUT when we detach we need to preserve the original aliasing information.

Differential Revision: [D64406859](https://our.internmc.facebook.com/intern/diff/D64406859/)

### CODE REVIEW

**Review by avikchaudhuri — CHANGES_REQUESTED** [review]
Test please

File: torch/export/_unlift.py [hunk 1]
```diff
@@ -309,6 +318,19 @@ def _create_stateful_graph_module(
     for const_name, value in ep.constants.items():
         if not torch.fx.graph_module._has_attr(stateful_gm, const_name):
             if isinstance(value, torch.Tensor):
+                if value.requires_grad:
+                    warnings.warn(
+                        f"A model attribute `{const_name}` requires gradient. "
```
Comment (avikchaudhuri) [inline torch/export/_unlift.py:323]: no ending period

File: torch/export/_unlift.py [hunk 2]
```diff
@@ -309,6 +318,19 @@ def _create_stateful_graph_module(
...
+                        f"but it's not properly registered as a parameter. "
+                        f"torch.export will detach it and treat it as a constant tensor "
+                        f"but please register it as parameter instead."
+                    )
+                    if value in original_tensor_to_detached_tensor:
+                        value = original_tensor_to_detached_tensor[value]
+                    else:
+                        detached_value = value.detach()
```
Comment (avikchaudhuri) [inline torch/export/_unlift.py:331]: maybe this code should be refactored into a util?

**Review by avikchaudhuri — APPROVED** [review]
Rebase and check CI before landing...

File: torch/export/_unlift.py (hunk 2 above)
Comment (tugsbayasgalan) [reply torch/export/_unlift.py:331]: It is bit hard to make it a util because this is only relevant in this specific location.

File: torch/export/_unlift.py (hunk 1 above)
Comment (tugsbayasgalan) [reply torch/export/_unlift.py:323]: sg

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D64406859).
